python/semantic_kernel/connectors/memory/astradb/astradb_memory_store.py

```diff
@@ -0,0 +1,300 @@
+            distance_type {str} -- Specifies the similarity metric to be used when querying or comparing vectors within this collection. The available options are dot_product, euclidean, and cosine.
+        Returns:
+            None
+        """
+        dimension_num = dimension_num if dimension_num is not None else self._embedding_dim
+        distance_type = distance_type if distance_type is not None else self._similarity
+
+        if dimension_num > MAX_DIMENSIONALITY:
+            raise ValueError(
+                f"Dimensionality of {dimension_num} exceeds "
+                + f"the maximum allowed value of {MAX_DIMENSIONALITY}."
+            )
+
+        result = await self._client.create_collection(
+            collection_name, dimension_num, distance_type)
+        if result == True:
+            self.logger.info(
+                f"Collection {collection_name} created.")
+
+    async def delete_collection_async(self, collection_name: str) -> None:
+        """Deletes a collection.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to delete.
+
+        Returns:
+            None
+        """
+        result = await self._client.delete_collection(collection_name)
+        if result == True:
+            self.logger.info(
+                f"Collection {collection_name} deleted.")
+        else:
+            self.logger.warning(
+                f"Collection {collection_name} does not exist.")
+
+    async def does_collection_exist_async(self, collection_name: str) -> bool:
+        """Checks if a collection exists.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to check.
+
+        Returns:
+            bool -- True if the collection exists; otherwise, False.
+        """
+        return await self._client.find_collection(collection_name)
+
+    async def upsert_async(self, collection_name: str, record: MemoryRecord) -> str:
+        """Upserts a memory record into the data store. Does not guarantee that the collection exists.
+            If the record already exists, it will be updated.
+            If the record does not exist, it will be created.
+
+        Arguments:
+            collection_name {str} -- The name associated with a collection of embeddings.
+            record {MemoryRecord} -- The memory record to upsert.
+
+        Returns:
+            str -- The unique identifier for the memory record.
+        """
+        filter = {"_id": record._id}
+        update = {"$set": build_payload(record)}
+        status = await self._client.update_document(
+            collection_name, filter, update, True)
+
+        return status["upsertedId"] if "upsertedId" in status else record._id
+
+    async def upsert_batch_async(
+        self, collection_name: str, records: List[MemoryRecord]
+    ) -> List[str]:
+        upserted_ids = []
+
+        for record in records:
+            id = await self.upsert_async(collection_name, record)
```

Comment:
could this be parallelized by using asyncio.gather?

Reply:
Done